src/components/ui/popover/type.ts

```diff
@@ -205,7 +205,7 @@ export class PopoverPlacements {
 
       return new Frame(
         origin.x,
-        origin.y - (options.other.size.height - size.height) / 2 + options.offsets.origin.y,
+        Math.round(origin.y - (options.other.size.height - size.height) / 2 + options.offsets.origin.y),
```

Comment:
Such behavior can repeats several times in code, i think we can make a function that will be performed in all necessary places.